samples/grids/tree-grid/data-searching/src/index.tsx

```diff
@@ -0,0 +1,152 @@
+const clearIconText =
+  "<svg width='24' height='24' viewBox='0 0 24 24' title='Clear'><path d='M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z'></path></svg>";
+
+const data = new EmployeesFlatData();
+
+export default function Sample() {
+  const gridRef = useRef<IgrTreeGrid>(null);
+  const clearIconRef = useRef<IgrIconButton>(null);
+  const iconButtonNextRef = useRef<IgrIconButton>(null);
+  const iconButtonPrevRef = useRef<IgrIconButton>(null);
+  const caseSensitiveChipRef = useRef<IgrChip>(null);
+  const exactMatchChipRef = useRef<IgrChip>(null);
+
+  const [searchText, setSearchText] = useState('')
+
+  useEffect(() => {
+    if (clearIconRef?.current) {
+      clearIconRef.current.registerIconFromText("clear", clearIconText, "material");
+    }
+    if (iconButtonPrevRef?.current) {
+      iconButtonPrevRef.current.registerIconFromText("prev", prevIconText,"material");
+    }
+    if (iconButtonNextRef?.current) {
+      iconButtonNextRef.current.registerIconFromText("next", nextIconText, "material");
+    }
+  }, []);
+
+  function handleOnSearchChange(input: IgrInput, event: IgrComponentValueChangedEventArgs) {
+    setSearchText(event.detail);
+    gridRef.current.findNext(event.detail, caseSensitiveChipRef.current.selected, exactMatchChipRef.current.selected);
+  }
+
+  function handleCaseSensitiveChange(chip: IgrChip, event: IgrComponentBoolValueChangedEventArgs) {
+    gridRef.current.findNext(searchText, event.detail, exactMatchChipRef.current.selected);
+  }
+
+  function handleExactMatchChange(chip: IgrChip, event: IgrComponentBoolValueChangedEventArgs) {
+    gridRef.current.findNext(searchText, caseSensitiveChipRef.current.selected, event.detail);
+  }
+
+  function clearSearch() {
+    setSearchText('');
+    gridRef.current.clearSearch();
+  }
+
+  function prevSearch() {
+    gridRef.current.findPrev(searchText, caseSensitiveChipRef.current.selected, exactMatchChipRef.current.selected);
+  }
+
+  function nextSearch() {
+    gridRef.current.findNext(searchText, caseSensitiveChipRef.current.selected, exactMatchChipRef.current.selected);
+  }
+
+  function searchKeyDown(e: KeyboardEvent<HTMLElement>) {
+    if (e.key === 'Enter' || e.key === 'ArrowDown') {
+        e.preventDefault();
+        gridRef.current.findNext(searchText, caseSensitiveChipRef.current.selected, exactMatchChipRef.current.selected);
+    } else if (e.key === 'ArrowUp') {
+        e.preventDefault();
+        gridRef.current.findPrev(searchText, caseSensitiveChipRef.current.selected, exactMatchChipRef.current.selected);
+    }
+  }
+
+  return (
+    <div className="container sample">
+      <div className="container vertical">
+        <div style={{ marginBottom: "1rem" }} onKeyDown={searchKeyDown}>
+          <IgrInput name="searchBox" value={searchText} inputOcurred={handleOnSearchChange}>
```

Comment:
Add checking which icon to show the search icon or clear icon based on whether there is a value entered in the IgrInput.

```tsx
{searchText.length === 0 ? (
                <IgrIconButton
                    key="searchIcon"
                    ref={searchIconRef} 
                    variant="flat"
                    name="search" 
                    collection="material"
                />
            ) : (
                <IgrIconButton
                    key="clearIcon"
                    ref={clearIconRef}
                    variant="flat"
                    name="clear"
                    collection="material"
                    clicked={clearSearch}
                />
            )}
```